src/Map/src/Bridge/Google/assets/src/map_controller.ts

```diff
@@ -0,0 +1,187 @@
+
+        return new library.Map(this.element, {
+            ...options,
+            center,
+            zoom,
+        });
+    }
+
+    protected doCreateMarker(
+        definition: MarkerDefinition<google.maps.marker.AdvancedMarkerElementOptions, google.maps.InfoWindowOptions>
+    ): google.maps.marker.AdvancedMarkerElement {
+        const { position, title, infoWindow, rawOptions = {}, ...otherOptions } = definition;
+
+        const marker = new library.AdvancedMarkerElement({
+            position,
+            title,
+            ...otherOptions,
+            ...rawOptions,
+            map: this.map,
+        });
+
+        if (infoWindow) {
+            this.createInfoWindow({ definition: infoWindow, marker });
+        }
+
+        return marker;
+    }
+
+    protected doCreateInfoWindow({
+        definition,
+        marker,
+    }: {
+        definition: MarkerDefinition<
+            google.maps.marker.AdvancedMarkerElementOptions,
+            google.maps.InfoWindowOptions
+        >['infoWindow'];
+        marker: google.maps.marker.AdvancedMarkerElement;
+    }): google.maps.InfoWindow {
+        const { headerContent, content, rawOptions = {}, ...otherOptions } = definition;
+
+        const infoWindow = new library.InfoWindow({
+            headerContent: this.createTextOrElement(headerContent),
+            content: this.createTextOrElement(content),
+            ...otherOptions,
+            ...rawOptions,
+        });
+
+        if (definition.opened) {
+            infoWindow.open({
+                map: this.map,
+                shouldFocus: false,
+                anchor: marker,
+            });
+        }
+
+        marker.addListener('click', () => {
+            if (definition.autoClose) {
+                this.closeInfoWindowsExcept(infoWindow);
+            }
+
+            infoWindow.open({
+                map: this.map,
+                anchor: marker,
+            });
+        });
+
+        return infoWindow;
+    }
+
+    private createTextOrElement(content: string | null): string | HTMLElement | null {
+        if (!content) {
+            return null;
+        }
+
+        // we assume it's HTML if it includes "<"
+        if (content.includes('<')) {
+            const div = document.createElement('div');
+            div.innerHTML = content;
+            return div;
+        }
+
+        return content;
+    }
+
+    private closeInfoWindowsExcept(infoWindow: google.maps.InfoWindow) {
+        this.infoWindows.forEach((otherInfoWindow) => {
+            if (otherInfoWindow !== infoWindow) {
+                otherInfoWindow.close();
+            }
+        });
+    }
+
+    protected doFitBoundsToMarkers(): void {
+        if (this.markers.length === 0) {
+            return;
+        }
+
+        const bounds = new google.maps.LatLngBounds();
+        this.markers.forEach((marker) => {
+            if (!marker.position) {
```

Comment:
Is there a real use case ?

Reply:
I don't think so, but `Marker#position` can be undefined or null in it's types declaration, so better playing it safe